graph-gateway/src/indexing_performance.rs

```diff
@@ -0,0 +1,155 @@
+use std::{collections::HashMap, ops::Deref, time::Duration};
+
+use alloy_primitives::{Address, BlockNumber};
+use parking_lot::RwLock;
+use thegraph_core::types::DeploymentId;
+use tokio::{
+    self,
+    sync::{mpsc, watch},
+    time::MissedTickBehavior,
+};
+
+use crate::network::{internal::NetworkTopologySnapshot, IndexingId};
+
+#[derive(Default)]
+pub struct Snapshot {
+    pub response: indexer_selection::Performance,
+    pub latest_block: Option<BlockNumber>,
+}
+
+#[derive(Clone)]
+pub struct IndexingPerformance {
+    data: &'static DoubleBuffer,
+    msgs: mpsc::UnboundedSender<Feedback>,
+}
+
+struct Feedback {
+    indexer: Address,
+    deployment: DeploymentId,
+    success: bool,
+    latency_ms: u16,
+    latest_block: Option<BlockNumber>,
+}
+
+impl IndexingPerformance {
+    pub fn new(network: watch::Receiver<NetworkTopologySnapshot>) -> Self {
```

Comment:
By passing this `watch::Receiver` to this service, we are coupling it with the `network` service as it will depend on the `NetworkTopologySnapshot` type. 

Could you generate a watch channel in the `network` service that only emits the indexing progress? Something that can be consumed here without requiring any processing:

```rust
pub fn new(network: watch::Receiver<HasMap<IndexingId, BlockNumber /*latest_block*/>>) -> Self {
```